src/libsystemd/sd-device/sd-device.c

```diff
@@ -1894,12 +1894,16 @@ _public_ int sd_device_get_sysattr_value(sd_device *device, const char *sysattr,
         path = prefix_roota(syspath, sysattr);
         r = lstat(path, &statbuf);
         if (r < 0) {
+                int k;
+
                 /* remember that we could not access the sysattr */
-                r = device_add_sysattr_value(device, sysattr, NULL);
-                if (r < 0)
-                        return r;
+                k = device_add_sysattr_value(device, sysattr, NULL);
+                if (k < 0)
+                        log_device_debug_errno(device, k,
+                                               "sd-device: failed to cache attribute '%s' with NULL, ignoring: %m",
+                                               sysattr);
```

Comment:
maybe flush out any old cached value if this fails? or does device_add_sysattr_value() guarantee that anyway?

Reply:
Yes. `device_add_sysattr_value()` first removes the old entry. So, we do not need to flush on failure.